server/player/player.go

```diff
@@ -2087,6 +2087,9 @@ func (p *Player) Collect(s item.Stack) int {
 	if p.Dead() {
 		return 0
 	}
+	if p.GameMode() == world.GameModeSpectator {
```

Comment:
Could you update this to use the `AllowsInteraction()` method from the GameMode interface? It's just a more generic way to achieve the same result and allows custom gamemodes to inherit the same functionality

Reply:
Ok